ada_feeding/scripts/ada_planning_scene.py

```diff
@@ -690,23 +694,50 @@ def update_table_detection(self) -> None:
         detected_table_pose.pose.position.y += self.table_detection_offsets[1]
         detected_table_pose.pose.position.z += self.table_detection_offsets[2]
 
-        # Calculate the angular distance between the latest and default table quaternions
-        latest_table_quat = np.array(
+        # Store default and latest table orientations as transforms3d quaternions
+        default_table_quat = np.array(
             [
+                self.objects[self.table_object_id].quat_xyzw[3],
+                self.objects[self.table_object_id].quat_xyzw[0],
+                self.objects[self.table_object_id].quat_xyzw[1],
+                self.objects[self.table_object_id].quat_xyzw[2],
+            ]
+        )
+        latest_quat_unrot = np.array(
+            [
+                detected_table_pose.pose.orientation.w,
                 detected_table_pose.pose.orientation.x,
                 detected_table_pose.pose.orientation.y,
                 detected_table_pose.pose.orientation.z,
-                detected_table_pose.pose.orientation.w,
             ]
         )
-        default_table_quat = np.array([0.0, 0.0, 1.0, 0.0]) # default quaternion with 180 degrees
-                                                            # rotation about the z-axis for more
-                                                            # intuitive calculation of distance
-        quat_dist = np.arccos(
-            2 * (np.dot(default_table_quat, latest_table_quat) ** 2) - 1
+
+        # Rotate the latest quaternion by 180 degrees across the z axis
+        # and store as a separate quaternion for comparison
+        z_axis_rotation = np.array([0.0, 0.0, 0.0, 1.0])
```

Comment:
Haha, this confused me at first since I interpreted it as x,y,z,w, so thought this was a 0 degree rotation. Until I realized it is w,x,y,z :) That's where a comment clarifying it would help.